core/templates/dev/head/pages/exploration_editor/EditorServices.js

```diff
@@ -997,19 +995,22 @@ oppia.factory('explorationStatesService', [
       saveFallbacks: function(stateName, newFallbacks) {
         saveStateProperty(stateName, 'fallbacks', newFallbacks);
       },
+      isInitialized: function() {
+        return _states != null;
+      },
       addState: function(newStateName, successCallback) {
         newStateName = $filter('normalizeWhitespace')(newStateName);
         if (!validatorsService.isValidStateName(newStateName, true)) {
           return;
         }
-        if (_states.hasOwnProperty(newStateName)) {
+        if (_states.hasState(newStateName)) {
           alertsService.addWarning('A state with this name already exists.');
           return;
         }
         alertsService.clearWarnings();
 
-        _states[newStateName] = newStateTemplateService.getNewStateTemplate(
```

Comment:
That's concerning, and we should not just sweep it under the carpet like this. I think the dependency should be removed if it is not being used. What is the Travis error exactly?

Also, is newStateTemplateService currently used anywhere apart from the StatesObjectFactory? If not, maybe we should get rid of it and have its functionality be internal to the StatesObjectFactory...